flow/project.py

```diff
@@ -1692,18 +1692,88 @@ def _store_bundled(self, operations):
             with open(fn_bundle, 'w') as file:
                 for operation in operations:
                     file.write(operation.id + '\n')
             return bid
 
+    def _store_aggregates(self, operations):
+        """Store aggregate-ids per operation information.
+
+        This enables status check of aggregates which were
+        formed previously but are not present currently for any
+        operation.
+
+        :param operations:
+            The operations to be submitted.
+        :type operations:
+            A sequence of instances of :py:class:`.JobOperation`
+        """
+        for operation in operations:
+            op_fn = '{}.txt'.format(operation.name)
+            aggregate_wid = '{} {} \n'.format(operation.id, ' '.join(map(str, operation._jobs)))
+            fn_aggregate = self._fn_stored('.aggregates', op_fn)
+            try:
+                os.makedirs(os.path.dirname(fn_aggregate), exist_ok=True)
+            except PermissionError:
+                continue
+            if os.path.exists(fn_aggregate):
+                with open(fn_aggregate, 'r+') as file:
+                    if aggregate_wid in file:
+                        continue
+                    else:
+                        file.write(aggregate_wid)
+            else:
+                with open(fn_aggregate, 'w') as file:
+                    file.write(aggregate_wid)
+
+    def _fetch_aggregates(self, group):
+        """Store aggregate-ids per operation information.
+
+        This enables status check of aggregates which were
+        formed previously but are not present currently for any
+        operation.
+
+        :param group:
+            FlowGroup associated with the operation.
+        :type group:
+            list :py:class:`flow.FlowGroup`
+        :return:
+            Fetched signac job handles that were previously submitted.
+        :rtype:
+            tuple
+        """
+        fetched_aggregates = []
+        dir = '.aggregates/{}.txt'.format(group.name)
+        if os.path.exists(dir):
+            with open(dir, 'r') as file:
+                for obj in file:
+                    aggregate = []
+                    _ids = obj.split(' ')
+                    submission_id = _ids[0]
+                    job_ids = _ids[1:-1]
+                    try:
+                        for job_id in job_ids:
+                            aggregate.append(self.open_job(id=job_id))
+
+                        aggregate = tuple(aggregate)
+                        # Checking whether the aggregate and the submission id match.
+                        # If not, then a user must have changed the submission id.
+                        # Hence skip this aggregate.
+                        if group._generate_id(aggregate) == submission_id:
+                            fetched_aggregates.append(aggregate)
+                    except KeyError:  # Not able to open the job via job id.
```

Comment:
What does it mean if we can't open the job via is when fetching aggregates? If we don't catch this error, what fails?

Reply:
A user may dynamically delete a job. And if a user does that, because the aggregate information is already stored in `.aggregates` directory, we'll definitely try to open it and catch an error.
So we skip that particular aggregate if we catch a `KeyError`

Reply:
I think this is the right approach.